msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/UIPermissionsManagerImpl.java

```diff
@@ -940,6 +941,34 @@ public boolean isModeratePostings(Long topicId, Boolean isForumLocked, Boolean i
     }
     return false;
   }
+
+  public boolean isIdentifyAnonAuthors(Topic topic)
+  {
+    String currentUserId = getCurrentUserId();
+    if (isSuperUser(currentUserId))
+    {
+      return true;
+    }
+
+    try
+    {
+      Iterator iter = getTopicItemsByUser(topic.getId(), currentUserId, getContextId());
+      while (iter.hasNext())
+      {
+        DBMembershipItem item = (DBMembershipItem) iter.next();
+        if (item.getPermissionLevel().getIdentifyAnonAuthors().booleanValue())
+        {
+          return true;
+        }
+      }
+    }
+    catch (Exception e)
+    {
+      LOG.error(e.getMessage(), e);
```

Comment:
What kind of exception are we going to get here? Should it be logged at error level?